app/src/androidTest/java/org/mozilla/fenix/ui/robots/DownloadRobot.kt

```diff
@@ -133,16 +133,36 @@ fun downloadRobot(interact: DownloadRobot.() -> Unit): DownloadRobot.Transition
 }
 
 private fun assertDownloadPrompt(fileName: String) {
-    assertTrue(
-        "Download prompt button not visible",
-        mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
-            .waitForExists(waitingTime)
-    )
-    assertTrue(
-        "$fileName title doesn't match",
-        mDevice.findObject(UiSelector().text(fileName))
-            .waitForExists(waitingTime)
-    )
+    var currentTries = 0
+    while (currentTries++ < 3) {
+        try {
+            assertTrue(
+                "Download prompt button not visible",
+                mDevice.findObject(UiSelector().resourceId("$packageName:id/download_button"))
+                    .waitForExists(waitingTime)
+            )
+            assertTrue(
+                "$fileName title doesn't match",
+                mDevice.findObject(UiSelector().text(fileName))
+                    .waitForExists(waitingTime)
+            )
+        } catch (e: AssertionError) {
+            browserScreen {
```

Comment:
It's good practice to log the `Assertion` here e.,g 
```Kotlin
println("Failed to find locator: ${e.localizedMessage}")
```

Reply:
Done!